### ISSUE TITLE
Subsources!

### ISSUE THREAD

**Issue by bertspaan** [histograph/data#17]
**Idea:**

Allow subsets per data source, for example `bag.streets`, `bag.buildings`, `tgn.places`, `tgn.terms`, et cetera. An owner can make as many sub sources, as long as he owns the parent source (or maybe subsources are created implicitly when adding PITs or relations to them). Users can query on parent sources (`source=bag` or `source=bag.*`), or on a single subsource (`source=bag.streets`).

**This is a good idea, because:**
- Some data sources consist of different types of data, with different meta data (in `data` field) per PIT
- Using subsources, Core itself can add a subsource for inferred relations (e.g. `tgn.places.inferred`)
- It's easy to update/delete a subset of your data, without touching the rest

@mmmenno, @phpetra, @wires what do you think?

This should be implemented in API, IO, Core and Viewer as well!

**Comment by mmmenno** [comment]
For gemeentegeschiedenis I had to make 2 datasets: 1 for the actual gemeentes and 1 for the geometries. I can imagine that doesn't make sense to users.

With tgn I was breaking my head over the question how to add foreign places without creating yet another dataset with a name like 'tgn-not-nl'. This idea sounds like just the solution I was looking for!

How much time will it take to make this work? Should we not concentrate on other issues first?

**Comment by bertspaan** [comment]
@mmmenno yes, we should concentrate on other issues first, but this is something we can slowly implement, we do not have to do it at once.

**Comment by mmmenno** [comment]
@bertspaan I'd say let's do it.
